tests/integration_tests/test_hyperopt.py

```diff
@@ -301,21 +338,122 @@ def test_hyperopt_run_hyperopt(csv_filename, search_space, tmpdir, ray_cluster):
-    if search_space == "random":
-        assert hyperopt_results.experiment_analysis.results_df.shape[0] == RANDOM_SEARCH_SIZE
-    else:
-        # compute size of search space for grid search
-        grid_search_size = 1
-        for k, v in search_parameters.items():
-            grid_search_size *= len(v["values"])
-        assert hyperopt_results.experiment_analysis.results_df.shape[0] == grid_search_size
+    with tempfile.TemporaryDirectory() as tmpdir:
+        hyperopt_results = hyperopt(config, dataset=rel_path, output_directory=tmpdir, experiment_name="test_hyperopt")
+        if search_space == "random":
+            assert hyperopt_results.experiment_analysis.results_df.shape[0] == RANDOM_SEARCH_SIZE
+        else:
+            # compute size of search space for grid search
+            grid_search_size = 1
+            for k, v in search_parameters.items():
+                grid_search_size *= len(v["values"])
+            assert hyperopt_results.experiment_analysis.results_df.shape[0] == grid_search_size
+
+        # check for return results
+        assert isinstance(hyperopt_results, HyperoptResults)
 
-    # check for return results
-    assert isinstance(hyperopt_results, HyperoptResults)
+        # check for existence of the hyperopt statistics file
+        assert os.path.isfile(os.path.join(tmpdir, "test_hyperopt", "hyperopt_statistics.json"))
 
-    # check for existence of the hyperopt statistics file
-    assert os.path.isfile(os.path.join(tmpdir, "test_hyperopt", "hyperopt_statistics.json"))
 
-    if os.path.isfile(os.path.join(tmpdir, "test_hyperopt", "hyperopt_statistics.json")):
-        os.remove(os.path.join(tmpdir, "test_hyperopt", "hyperopt_statistics.json"))
+@pytest.mark.distributed
+@pytest.mark.parametrize("search_space", ["random"])
+def test_hyperopt_run_hyperopt_with_shared_params(csv_filename, search_space):
+    config, rel_path = _setup_ludwig_config_with_shared_params(csv_filename)
+
+    categorical_feature_name = config[INPUT_FEATURES][2][NAME]
+    output_feature_name = config[OUTPUT_FEATURES][0][NAME]
+
+    cell_types_search_space = ["lstm", "gru"]
+    vocab_size_search_space = list(range(4, 9))
+
+    # Create default search space for text features with various cell types
+    search_parameters = {
+        "trainer.learning_rate": {
+            "lower": 0.0001,
+            "upper": 0.01,
+            "space": "loguniform",
+        },
+        categorical_feature_name + ".vocab_size": {"space": "randint", "lower": 1, "upper": 3},
+        DEFAULTS
+        + "."
+        + INPUT_FEATURES
+        + "."
+        + TEXT
+        + ".cell_type": {"space": "choice", "categories": cell_types_search_space},
+        DEFAULTS + "." + OUTPUT_FEATURES + "." + CATEGORY + ".vocab_size": {"space": "randint", "lower": 4, "upper": 8},
+    }
+
+    # add hyperopt parameter space to the config
```

Comment:
@justinxzhao I had done that initially, but in a prior code review for this PR, there was a suggestion to make this part out into a separate function. Any strong reasons beyond having all the code in one place to make it easier to read to move this all into one function? 

I think the main reason to have this in a separate function is that in case additional tests need to be added in the future using the same config, it makes sense to have it be returned programmatically rather than repeating code.

Reply:
Here, my suggestion is just to make the config["hyperopt"] expressed as one block instead of multiple blocks, unless we're doing heavy parameterization, which I don't think this test necessarily needs.

```python
@pytest.mark.distributed
def test_hyperopt_run_hyperopt_with_shared_params(csv_filename, tmpdir):
    config, rel_path = _setup_ludwig_config_with_shared_params(csv_filename)

    categorical_feature_name = config[INPUT_FEATURES][2][NAME]
    output_feature_name = config[OUTPUT_FEATURES][0][NAME]

    cell_types_search_space = ["lstm", "gru"]
    vocab_size_search_space = list(range(4, 9))

    config["hyperopt"] = {
        "parameters": {
            "trainer.learning_rate": {"lower": 0.0001, "upper": 0.01, "space": "loguniform"},
            f"{categorical_feature_name}.vocab_size": {"space": "randint", "lower": 1, "upper": 3},
            "defaults.input_features.text.cell_type": {"space": "choice", "categories": ["lstm", "gru"]},
            "defaults.output_features.category.vocab_size": {"space": "randint", "lower": 4, "upper": 8},
        },
        "goal": "minimize",
        "output_feature": output_feature_name,
        "validation_metrics": "loss",
        "executor": {"type": "ray", "num_samples": 4},
        "search_alg": {"type": "variant_generator"},
    }

    hyperopt_results = hyperopt(config, dataset=rel_path, output_directory=tmpdir, experiment_name="test_hyperopt")

    # Check that the trials did sample from defaults in the search space
    ...
```

Reply:
I see an issue here. vocab size is an internal parameter, not an externally visible one (we should find a better way to make the distinction clear).
What you actually want to use .preprocessing.most_common (also not a great name, we may want to change it in the future to max_vocab_size)